### PR TITLE
:new: :mortar_board: Unit Testing Collections Topic --- Test singleton stack

### PR DESCRIPTION
**Pull request by jameshughes89**
### Related Issues or PRs
#690 

### What
- Explain singleton tests
- Explain nested classes
- Note that the push/newTop tests are useless 

### Testing
:+1: built local, looks good

### CODE REVIEW

File: src/site/topics/testing/unit-test-collections.rst [hunk 1]
```diff
@@ -158,6 +158,206 @@ Common Setup Code
...
+        classUnderTest = new ArrayStack<>();
+        preState = new ArrayStack<>();
+    }
+
+    // Empty tests excluded here
+
+    @Test
+    void push_singleton_returnsTrue() {
+        classUnderTest.push(10);
+        assertTrue(classUnderTest.push(11));
```
Comment (twentylemon) [inline src/site/topics/testing/unit-test-collections.rst:188]: despite not being used, `preState` should be updated here as well -- really hammers home (imo) why `@BeforeEach` and `@Nested` are simplifications and useful.

File: src/site/topics/testing/unit-test-collections.rst [hunk 2]
```diff
@@ -158,6 +158,206 @@ Common Setup Code
...
+
+
+    This highlights the complexities caused by the *interconnectedness* of the collection's methods --- one cannot test
+    that ``push`` results in a new top without using ``peek``, and one cannot test ``peek`` without having already
+    called ``push``.
+
+
+
+Nested Classes
+--------------
```
Comment (twentylemon) [inline src/site/topics/testing/unit-test-collections.rst:275]: _Nested **Test** Classes_  -- just to be specific, nested classes are introduced with linked implementations later.

**Review by twentylemon — APPROVED** [review]

File: src/site/topics/testing/unit-test-collections.rst (hunk 1 above)
Comment (jameshughes89) [reply src/site/topics/testing/unit-test-collections.rst:188]: I'm conflicted on this. I agree that it certainly motivates the `@BeforeEach` value, but it's also not used at all in any test but one. It's in the many tests where it's used again, which is the next section. 

I'll think on it. It's kinda' a funny, peculiar situation.

Comment (jameshughes89) [reply src/site/topics/testing/unit-test-collections.rst]: I ended up adding it with a note. I think the note helps makes it make more sense why it's a class field and always set to be equal to `classUnderTest`.

File: src/site/topics/testing/unit-test-collections.rst [hunk 3]
```diff
@@ -167,6 +167,14 @@ Singleton Case Stack Tests
     * To verify this, one can assert equality between ``preState`` and ``classUnderTest`` after calling ``peek`` on ``classUnderTest``
 
 
+.. note::
+
+    Although ``preState`` is only actually used on one of the tests so far, it is convenient to always have an
+    equivalent object to ``classUnderTest`` within the test class. For this reason, ``preState`` is made a class field
+    and is always setup the same way as ``classUnderTest``.
```
Comment (jameshughes89) [inline src/site/topics/testing/unit-test-collections.rst:174]: Note referenced in previous comment.

**Review by twentylemon — APPROVED** [review]

**Pull request merged**